theano/tensor/nnet/nnet.py

```diff
@@ -636,14 +638,17 @@ def local_softmax_with_bias(node):
                     vector_sum = tensor.add(*vectors)
                 else:
                     vector_sum = vectors[0]
+                copy_stack_trace(x_in, vector_sum)
```

Comment:
I would prefer to call `copy_stack_trace` only if a new `add` node is created, because otherwise `vectors[0]` already has a stack trace. Could you put it inside the `if len(...)` block?